agents/builtin/mysql/slowlog/slowlog.go

```diff
@@ -175,17 +176,20 @@ func (m *SlowLog) Run(ctx context.Context) {
 			res := aggregator.Finalize()
 
 			// Check if MySQL SlowLog config is changed and slowlog rotated.
-			curStat, err := os.Stat(slowLogFilePath)
+			newSlowLogFileStat, err := os.Stat(slowLogFilePath)
 			if err != nil {
 				m.l.Errorf("cannot get stat of slowlog (%s): %s", slowLogFilePath, err)
 				return
 			}
-			if !os.SameFile(stat, curStat) {
-				opts.StartOffset = uint64(curStat.Size())
+			// In case of rotatation a slowlog file, set a new offset to parse the new slow log file.
+			if !os.SameFile(operatingSlowLogFileStat, newSlowLogFileStat) {
+				opts.StartOffset = uint64(newSlowLogFileStat.Size())
+				// use new slowlog stat as current.
+				operatingSlowLogFileStat = newSlowLogFileStat
 			}
 
 			if !running {
-				m.changes <- Change{Status: inventorypb.AgentStatus_STARTING}
+				m.changes <- Change{Status: inventorypb.AgentStatus_RUNNING}
```

Comment:
`STARTING` is for when we started doing something, but not received any data yet. Looks like STARTING is a correct value there.

Reply:
Yeah, agree.